src/decoderbufs.c

```diff
@@ -670,7 +670,7 @@ static void pg_decode_change(LogicalDecodingContext *ctx, ReorderBufferTXN *txn,
   // set primary key only if it exists for the table
   TupleDesc index_tupdesc = NULL;
   Relation index_rel = NULL;
-  if (!is_rel_non_selective) {
+  if (!is_rel_non_selective && OidIsValid(relation->rd_replidindex)) {
```

Comment:
it seems there are other types of replica identities when is_rel_non_selective is false, but still rd_replidindex is NULL.

So, we just cannot open null rd_replidindex, and it does not make sense even to try in this case

Reply:
PS:  After merge, we will see in p2q what kind of tables are these (that does not have rd_replidindex).